packages/edit-site/src/components/dataviews/sidebar-content.js

```diff
@@ -1,4 +1,205 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalItemGroup as ItemGroup,
+	DropdownMenu,
+	MenuGroup,
+	MenuItem,
+	Modal,
+	Button,
+	TextControl,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { moreVertical, plus, page, columns } from '@wordpress/icons';
+import { __ } from '@wordpress/i18n';
+import { useContext, useState, useEffect, useMemo } from '@wordpress/element';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import SidebarNavigationItem from '../sidebar-navigation-item';
+import DataviewsContext from '../dataviews/context';
+import { DEFAULT_VIEWS } from './provider';
+
+function DataViewItem( { dataviewId, setCurrentViewId, isActive } ) {
+	const { dataview } = useSelect(
+		( select ) => {
+			const { getEditedEntityRecord } = select( coreStore );
+			return {
+				dataview: getEditedEntityRecord(
+					'postType',
+					'wp_dataviews',
+					dataviewId
+				),
+			};
+		},
+		[ dataviewId ]
+	);
+	const icon = useMemo( () => {
+		const viewContent = JSON.parse( dataview.content );
+		switch ( viewContent.type ) {
+			case 'list':
+				return page;
+			case 'grid':
+				return columns;
+		}
+	}, [ dataview.content ] );
+	return (
+		<SidebarNavigationItem
+			icon={ icon }
+			onClick={ () => {
+				setCurrentViewId( dataviewId );
+			} }
+			aria-current={ isActive ? 'true' : undefined }
+			suffix={
+				<DropdownMenu
+					icon={ moreVertical }
+					label={ __( 'Actions' ) }
+					toggleProps={ {
+						style: {
+							color: 'inherit',
+						},
+					} }
+				>
+					{ ( { onClose } ) => (
+						<MenuGroup>
+							<MenuItem onClick={ () => onClose() }>
+								{ __( 'Rename' ) }
+							</MenuItem>
+							<MenuItem onClick={ () => onClose() }>
+								{ __( 'Delete' ) }
+							</MenuItem>
+						</MenuGroup>
+					) }
+				</DropdownMenu>
+			}
+		>
+			{ dataview.title }
+		</SidebarNavigationItem>
+	);
+}
+
+function AddNewItem( { type, taxonomyId } ) {
+	const { saveEntityRecord } = useDispatch( coreStore );
+
+	const [ isAdding, setIsAdding ] = useState( false );
+	const [ title, setTitle ] = useState( '' );
+	const { dataViewIsSaving } = useSelect( ( select ) => {
+		const { isSavingEntityRecord } = select( coreStore );
+		return {
+			dataViewIsSaving: isSavingEntityRecord(
+				'postType',
+				'wp_dataviews'
+			),
+		};
+	}, [] );
+	useEffect( () => {
+		if ( ! dataViewIsSaving ) {
+			setIsAdding( false );
+		}
+	}, [ dataViewIsSaving ] );
+	useEffect( () => {
```

Comment:
Can we avoid this effect by just calling setTitle directly when needed? (Yeah, if we can avoid useEffect, we'll avoid subtle bugs :P )

Reply:
We can do that, but in this case,   every time the modal closes, I want to clean the title. It seemed better to have the effect that calls setTitle in every place that closes the modal. This seemed more future-proof as soon we may have a place that closes the modal and we forget to reset the title. But either option is fine I will remove the effect,

Reply:
Maybe extract the Modal code in a different component so it will keep its state and reset when closed/reopened?